lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -238,12 +239,14 @@ public SystemAccessControl create(Map<String, String> config)
                 return ForwardingSystemAccessControl.of(memoizeWithExpiration(
                         () -> {
                             log.info("Refreshing system access control from %s", configFileName);
-                            return create(configFileName);
+                            FileBasedSystemAccessControlRules rules = injector.getInstance(FileBasedSystemAccessControlRules.class);
```

Comment:
Instead of using `injector` - can we pass a `Supplier` here.

Reply:
refactored